app/ota_boot.py

```diff
@@ -1,78 +1,201 @@
 #!/usr/bin/env python3
-
-import os
+from abc import ABC, abstractmethod
+from pathlib import Path
 import shutil
 
-import ota_status
-import grub_control
+from ota_status import OtaStatus
+from grub_control import GrubCtl
+from constants import OtaBootStatusString, OtaStatusString
+from exceptions import OtaBootError
+import configs as cfg
 
 from logging import getLogger, INFO, DEBUG
 
 logger = getLogger(__name__)
-logger.setLevel(INFO)
+logger.setLevel(cfg.LOG_LEVEL_TABLE.get(__name__, cfg.DEFAULT_LOG_LEVEL))
+
+
+class OtaBootInterface(ABC):
+    """
+    OtaBoot interface for implementing OtaBoot
+    """
+
+    # alias
+    require_check, bypass_check = True, False
+    check_passed, check_failed = True, False
+    finalize_succeeded, finalize_failed = True, False
+    bypass_finalization_check_passed, bypass_finalization_check_failed = (
+        "no_f_c_p",
+        "no_f_c_f",
+    )
+    bypass_check = "no_c"
```

Comment:
What is those alias? Configuration? There are two `bypass_check`.

Reply:
Fix in commit cd00535.